### PR TITLE
Fix corruption in header parsing with Julia 0.6

### PR DESCRIPTION
**Pull request by wookay**
Hello, 
to fix the issue https://github.com/JuliaWeb/HttpServer.jl/issues/111

thanks.

### CODE REVIEW

**Comment by tanmaykm** [comment]
Thanks. I can confirm that I don't see the issue in #111 anymore with this fix.

**Comment by tanmaykm** [comment]
Can we merge this? @malmaud or @quinnj ?

File: src/RequestParser.jl [hunk 1]
```diff
@@ -48,24 +48,37 @@ end
...
+    header_value = s[1:len]
+    par.vec_values[par.num_values] = string(par.vec_values[par.num_values], header_value)
     return 0
 end
 
 function on_headers_complete(parser)
-    r = pd(parser).request
+    par = pd(parser)
+    r = par.request
+    merge!(r.headers, Dict(zip(par.vec_fields, par.vec_values)))
```
Comment (quinnj) [inline src/RequestParser.jl]: Note that this isn't strictly correct. By calling `zip` on your `par.vec_fields`, you're inadvertently dropping any duplicate header fields, such as `Set-Cookie`, which is commonly sent as several separate header fields. Not that this was being handled correctly before or necessarily needs to be addressed here, but it was something I remember fixing in the port of everything to HTTP.jl.

**Comment by tanmaykm** [comment]
Thanks @quinnj . This was blocking some of my packages. I will also tag a new release.

Planning to migrate to HTTP.jl soon.

**Pull request merged**